apps/src/javalab/JavalabFileHelper.js

```diff
@@ -18,20 +18,80 @@ export function getDefaultFileContents(filename, viewMode) {
 export const getTabKey = index => `file-${index}`;
 
 export const fileMetadataForEditor = (sources, isEditingStartSources) => {
+  /*
+  'sources' contains file information as key-value pairs.
+  The key is the file name such as 'Class1.java'.
+  The value is a file object with the keys: text (source code),
+  tabOrder (order of the file tab in orderedTabs from 0 to (number of files - 1)),
+  isVisible, and isValidation (default is false)
+  The orderOfFiles array will contain the 'fileOrder' of each file as they are assigned
+  their tabKey names (such as 'file-0', 'file-1') which is stored in fileMetadata.
+  For example, if sources contains the following:
+  {
+    'Class1.java': {text: '', tabOrder: 2, isVisible: true},
+    'Class2.java': {text: '', tabOrder: 0, isVisible: true},
+    'Class3.java': {text: '', tabOrder: 1, isVisible: true}
+  },
+  then depending on how Object.keys iterates through the keys on sources, we could have
+  fileMetadata assigned: {'file-0': 'Class2.java', 'file-1': 'Class1.java', 'file-2': 'Class3.java'}.
```

Comment:
I'm not sure I understand this comment. Wouldn't it be 
`{'file-0': 'Class2.java', 'file-1': 'Class3.java', 'file-2': 'Class1.java'}`?

Reply:
In line 36, {'file-0': 'Class2.java', 'file-1': 'Class1.java', 'file-2': 'Class3.java'} is fileMetadata which is assigned as Object.keys iterates through the keys (file names) in sources and assigns the tabKey (metadata) names 'file-0', 'file-1', and 'file-2'. I chose an arbitrary order of keys within sources. 
Also while Object.keys iterates through the keys in sources, the orders of each file are added in array orderOfFiles while tabKey names are added to array unorderedTabKeys. 
From both of these arrays, we then get the orderedTabKeys. 
I was considering using the file names instead of the tabKey names for the ordering. However, the tabKeys are already being used in multiple Java Lab files.

Reply:
Update: tabKeyNames are assigned to orderedTabKeys as we iterate through keys in sources, removing need for unorderedTabKeys - see [Sanchit's suggestion](https://github.com/code-dot-org/code-dot-org/pull/48183#discussion_r978868244) below.